test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -1225,26 +1222,22 @@ private LogReplicationSourceManager setupSourceManagerAndObservedValues(Set<Stri
 
         // Add this class as observer of the value of interest for the wait condition
         for (WAIT waitCondition : waitConditions) {
-            switch(waitCondition) {
-                case ON_ACK:
-                case ON_ACK_TS:
-                    ackMessages = sourceDataSender.getAckMessages();
-                    ackMessages.addObserver(this);
-                    break;
-                case ON_ERROR: // Wait on Error Notifications to Source
-                case ON_TIMEOUT_ERROR:
-                    errorsLogEntrySync = sourceDataSender.getErrors();
-                    errorsLogEntrySync.addObserver(this);
-                    break;
-                case ON_METADATA_RESPONSE:
-                    metadataResponseObservable = sourceDataSender.getMetadataResponses();
-                    metadataResponseObservable.addObserver(this);
-                case ON_SINK_RECEIVE: // Wait on Received Messages on Sink (Destination)
-                    sinkReceivedMessages = sourceDataSender.getSinkManager().getRxMessageCount();
-                    sinkReceivedMessages.addObserver(this);
-                default:
-                    // Nothing
-                    break;
+            if (waitCondition == WAIT.ON_ACK || waitCondition == WAIT.ON_ACK_TS) {
+                ackMessages = sourceDataSender.getAckMessages();
+                ackMessages.addObserver(this);
+            } else if (waitCondition == WAIT.ON_ERROR || waitCondition == WAIT.ON_TIMEOUT_ERROR) {
+                // Wait on Error Notifications to Source
+                errorsLogEntrySync = sourceDataSender.getErrors();
+                errorsLogEntrySync.addObserver(this);
+            } else if (waitCondition == WAIT.ON_METADATA_RESPONSE) {
+                metadataResponseObservable = sourceDataSender.getMetadataResponses();
+                metadataResponseObservable.addObserver(this);
+            }
+
+            if (waitCondition == WAIT.ON_METADATA_RESPONSE || waitCondition == WAIT.ON_SINK_RECEIVE) {
+                // Wait on Received Messages on Sink (Destination)
+                sinkReceivedMessages = sourceDataSender.getSinkManager().getRxMessageCount();
+                sinkReceivedMessages.addObserver(this);
```

Comment:
I think it could improve the readability a little bit by this:
```
            } else if (waitCondition == WAIT.ON_METADATA_RESPONSE) {
                metadataResponseObservable = sourceDataSender.getMetadataResponses();
                metadataResponseObservable.addObserver(this);
                // Wait on Received Messages on Sink (Destination)
                sinkReceivedMessages = sourceDataSender.getSinkManager().getRxMessageCount();
                sinkReceivedMessages.addObserver(this);
            } else if (waitCondition == WAIT.ON_SINK_RECEIVE) {
                // Wait on Received Messages on Sink (Destination)
                sinkReceivedMessages = sourceDataSender.getSinkManager().getRxMessageCount();
                sinkReceivedMessages.addObserver(this);
```

Reply:
Done, Thank you.